src/wifi_rssi_sensor.rs

```diff
@@ -52,10 +52,10 @@ impl SensorT<f64> for WifiRSSISensor {
         log::debug!("wifi-rssi sensor - get readings called");
         let mut ap_info = wifi_ap_record_t::default();
         unsafe {
-            match esp_wifi_sta_get_ap_info(&mut ap_info as *mut wifi_ap_record_t) {
+            let res = esp_wifi_sta_get_ap_info(&mut ap_info as *mut wifi_ap_record_t);
+            match res {
                 ESP_OK => {}
-                // TODO(RSDK-7801): Support #from EspError for SensorError
-                err => return Err(SensorError::SensorCodeError(err)), //TODO: impl #from EspError variant
+                _err => return Err(SensorError::EspError(EspError::from(res).expect("Failed to create EspError"))), 
```

Comment:
The `expect` here suggests that we could panic if `EspError::from` failed, but I don't think it really can. Is there another `EspError::` function you could make use of here?

Reply:
Maybe you can use `https://docs.esp-rs.org/esp-idf-sys/esp_idf_sys/struct.EspError.html#method.from_infallible` since the compiler should be able to see statically that `res` cannot be `ESP_OK` here?

Reply:
There's also a convert method that can let you turn this whole match statement into a normal result 

https://docs.esp-rs.org/esp-idf-svc/esp_idf_svc/hal/sys/struct.EspError.html#method.convert